src/route-href.js

```diff
@@ -0,0 +1,40 @@
+import {customAttribute, bindable} from 'aurelia-templating';
+import {inject} from 'aurelia-dependency-injection';
+import {Router} from 'aurelia-router';
+import {TaskQueue} from 'aurelia-task-queue';
+
+@customAttribute('route-href')
+@bindable({name: 'route', changeHandler: 'processChange'})
+@bindable({name: 'params', changeHandler: 'processChange'})
+@bindable({name: 'attribute', defaultValue: 'href'})
+@inject(Router, Element, TaskQueue)
+export class RouteHref {
+  constructor(router, element, taskQueue) {
+    this.router = router;
+    this.element = element;
+    this.taskQueue = taskQueue;
+    this.isChangePending = false;
+  }
+
+  attributeChanged(value, previous) {
+    if (previous) {
+      this.element.removeAttribute(previous);
+    }
+
+    this.processChange();
+  }
+
+  processChange() {
+    if (!this.isChangePending) {
+      this.isChangePending = true;
+      this.taskQueue.queueMicroTask(this.updateAttribute.bind(this));
+    }
+  }
+
+  updateAttribute() {
+    this.isChangePending = false;
+
+    let href = this.router.generate(this.route, this.params);
```

Comment:
Noted above,

> This is just to coalesce changes from the 3 bindable change handlers that fire with initial values.

The 3 change handlers all fire with initial values, and I only wanted to do the work once when all values had been set. Is there a simpler way to accomplish this?